### PR TITLE
First Prometheus Operator Chart

### PR DESCRIPTION
**Pull request by pipo02mix**
Create the first version of Prometheus operator chart

### CODE REVIEW

File: helm/kubernetes-prometheus-chart/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+description: A Helm Chart for Prometheus Operator.
+icon: https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/assets/prometheus_logo-cb55bb5c346.png
+name: kubernetes-prometheus-chart
```
Comment (sslavic) [inline helm/kubernetes-prometheus-chart/Chart.yaml]: Chart name is prefixed with repo name so https://github.com/giantswarm/architect-orb/blob/master/src/commands/package-and-push.yaml#L12 check should pass.

@kopiczko is "-chart" suffix supported in architect / architect-orb?

File: helm/kubernetes-prometheus-chart/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+description: A Helm Chart for Prometheus Operator.
+icon: https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/assets/prometheus_logo-cb55bb5c346.png
```
Comment (rossf7) [inline helm/kubernetes-prometheus-chart/Chart.yaml]: <3

File: helm/kubernetes-prometheus-chart/templates/NOTES.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+The Prometheus Operator has been installed. Check its status by running:
+  kubectl --namespace {{ .Release.Namespace }} get pods -l "release={{ .Release.Name }}"
+
+Visit https://github.com/giantswarm/prometheus-operator for instructions on how
```
Comment (rossf7) [inline helm/kubernetes-prometheus-chart/templates/NOTES.txt]: Not sure on this URL. It doesn't exist and I don't think this is a good name. See my main comment.

File: helm/kubernetes-prometheus-chart/templates/prometheus-operator/clusterrole.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  - list
+  - watch
+- apiGroups:
+  - ""
+  resources:
+  - namespaces
+  verbs:
+  - get
+  - list
+  - watch
```
Comment (rossf7) [inline helm/kubernetes-prometheus-chart/templates/prometheus-operator/clusterrole.yaml]: Not sure. Is this line ending a problem?

File: helm/kubernetes-prometheus-chart/templates/prometheus-operator/crd-alertmanager.yaml [hunk 5]
```diff
@@ -0,0 +1,2401 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (rossf7) [inline helm/kubernetes-prometheus-chart/templates/prometheus-operator/crd-alertmanager.yaml]: This CRD is really big. For my interest where does it come from?

Should the README mention which upstream repo this is based on?

File: helm/kubernetes-prometheus-chart/values.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+name: prometheus-operator
+namespace: kube-system
```
Comment (rossf7) [inline helm/kubernetes-prometheus-chart/values.yaml]: Just a question. Should the operator run in `kube-system` or in `giantswarm`? As this is where chart-operator runs.

File: helm/kubernetes-prometheus-chart/values.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  repository: quay.io/giantswarm/prometheus-operator
+  tag: v0.30.1
+
+prometheusConfigReloaderImage:
+  repository: quay.io/coreos/prometheus-config-reloader
+  tag: v0.30.1
+
+configmapReloadImage:
+  repository: quay.io/coreos/configmap-reload
+  tag: v0.0.1
```
Comment (rossf7) [inline helm/kubernetes-prometheus-chart/values.yaml]: Again line ending is weird.

File: helm/kubernetes-prometheus-chart/Chart.yaml (hunk 1 above)
Comment (rossf7) [reply helm/kubernetes-prometheus-chart/Chart.yaml]: @sslavic The `-chart` suffix isn't considered anymore. The validation was loosened by Pawel here. https://github.com/giantswarm/architect/pull/316

See my main comment for what I think we should use for the repo and chart names.

File: helm/kubernetes-prometheus-chart/values.yaml (hunk 6 above)
Comment (kopiczko) [reply helm/kubernetes-prometheus-chart/values.yaml]: I vote for `giantswarm`.

File: helm/kubernetes-prometheus-chart/Chart.yaml (hunk 1 above)
Comment (pipo02mix) [reply helm/kubernetes-prometheus-chart/Chart.yaml]: but then it is ok. If we want to make `helm lint` work too the folder and chart name should match

File: helm/kubernetes-prometheus-chart/values.yaml (hunk 6 above)
Comment (pipo02mix) [reply helm/kubernetes-prometheus-chart/values.yaml]: yeah me too, I change the default to `giantswarm` (the home of operators)

File: helm/kubernetes-prometheus-chart/templates/prometheus-operator/crd-alertmanager.yaml (hunk 5 above)
Comment (pipo02mix) [reply helm/kubernetes-prometheus-chart/templates/prometheus-operator/crd-alertmanager.yaml]: yeah let's add it. Yeah it comes from upstream

File: helm/kubernetes-prometheus-chart/Chart.yaml (hunk 1 above)
Comment (rossf7) [reply helm/kubernetes-prometheus-chart/Chart.yaml]: I think we can make them match. I've had an idea on how we should do the naming. I'm going to write it up in a discuss ticket later today.

So for now, let's just get this working and not worry about it too much.

File: helm/kubernetes-prometheus-chart/values.yaml (hunk 6 above)
Comment (rossf7) [reply helm/kubernetes-prometheus-chart/values.yaml]: Yes, I really like using `giantswarm`. 👍

**Comment by pipo02mix** [comment]
Added suggestions. Naming will be discussed appart

**Pull request merged**